File: src/immersed.dive.shop.model/Course.cs
using System;
using System.Collections.Generic;

namespace immersed.dive.shop.model
{
    public class Course : IEntity
    {
        public Course()
        {
            Id = Guid.NewGuid();

            Events = new List<Event>();
        }

        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }

        public List<Event> Events{ get; set; }

        public DateTime DateCreated { get; set; }
        public DateTime LastUpdated { get; set; }
        public bool Live { get; set; }
    }
}

File: the-client-app/src/app/models/course.ts

export interface Course {
    id?: string;    
    name?: string;
    description?: string;
    dateCreated?: Date;
    lastUpdated?: Date;
    live?:boolean;
}

export class Course implements Course {
    constructor(course?:Course) {
        if(course) {
            this.id = course.id;
            this.name = course.name;
            this.description = course.description;
            this.dateCreated = course.dateCreated;
            this.lastUpdated = course.lastUpdated;
            this.live = course.live;
        }
    }
}